engine/proofview.ml

```diff
@@ -345,22 +345,12 @@ let tclBREAK = Proof.break
 
 exception NoSuchGoals of int
 
-(* This hook returns a string to be appended to the usual message.
-   Primarily used to add a suggestion about the right bullet to use to
-   focus the next goal, if applicable. *)
-let nosuchgoals_hook:(int -> Pp.t) ref = ref (fun n -> mt ())
-let set_nosuchgoals_hook f = nosuchgoals_hook := f
-
-
-
 (* This uses the hook above *)
 let _ = CErrors.register_handler begin function
   | NoSuchGoals n ->
-    let suffix = !nosuchgoals_hook n in
-    CErrors.user_err 
-      (str "No such " ++ str (String.plural n "goal") ++ str "." ++
-       pr_non_empty_arg (fun x -> x) suffix)
-      | _ -> raise CErrors.Unhandled
+    CErrors.user_err
+      (str "No such " ++ str (String.plural n "goal") ++ str ".")
```

Comment:
But more generally, it seems wrong to have an handler for `NoSuchGoals` if the goal is to always catch this exception and re-raise it as `SuggestNoSuchGoal`.

Reply:
Hum, OK I see. There are cases where the exception is not supposed to be caught and this corresponds to previous cases where the hook was put back to the default value. Not the easiest code to read indeed :laughing:

Reply:
Indeed these codepaths are a bit tricky, they are getting a bit better after #9566 I hope.